pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -297,14 +297,14 @@ func (f *framework) RunPreFilterPlugins(
 	return nil
 }
 
-// RunPreFilterUpdaterAddPod calls the AddPod interface for the set of configured
+// RunPreFilterExtensionAddPod calls the AddPod interface for the set of configured
 // PreFilter plugins. It returns directly if any of the plugins return any
 // status other than Success.
-func (f *framework) RunPreFilterUpdaterAddPod(pc *PluginContext, podToSchedule *v1.Pod,
+func (f *framework) RunPreFilterExtensionAddPod(pc *PluginContext, podToSchedule *v1.Pod,
 	podToAdd *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *Status {
 	for _, pl := range f.preFilterPlugins {
-		if updater := pl.Updater(); updater != nil {
-			status := updater.AddPod(pc, podToSchedule, podToAdd, nodeInfo)
+		if extensions := pl.Extensions(); extensions != nil {
```

Comment:
may be you don't even need to assign to a different variable, you can do something like this in the for loop to reduce indentation:

```
if pl.Extensions() == nil {
 continue;
}

if status := extension.RemovePod(pc, podToSchedule, podToRemove, nodeInfo); !status.IsSuccess() {
  msg := fmt.Sprintf("error while running RemovePod for plugin %q while scheduling pod %q: %v",
                     pl.Name(), podToSchedule.Name, status.Message())
  klog.Error(msg)
  return NewStatus(Error, msg)
}
```

Reply:
Thanks, very good suggestion~